tests/core/test_memory.py

```diff
@@ -81,33 +121,39 @@ def test_empty_model_summary_shapes(mode):
     assert summary.param_nums == []
 
 
-@pytest.mark.parametrize(['mode'], [
-    pytest.param(ModelSummary.MODE_FULL),
-    pytest.param(ModelSummary.MODE_TOP),
-])
-@pytest.mark.parametrize(['device'], [
-    pytest.param(torch.device('cpu')),
-    pytest.param(torch.device('cuda', 0)),
-    pytest.param(torch.device('cuda', 0)),
-])
+@pytest.mark.parametrize(
+    ["mode"],
+    [
+        pytest.param(ModelSummary.MODE_FULL),
+        pytest.param(ModelSummary.MODE_TOP),
+    ],
+)
+@pytest.mark.parametrize(
+    ["device"],
+    [
+        pytest.param(torch.device("cpu")),
+        pytest.param(torch.device("cuda", 0)),
+        pytest.param(torch.device("cuda", 0)),
+    ],
+)
 @pytest.mark.skipif(not torch.cuda.is_available(), reason="Test requires GPU.")
```

Comment:
The `cuda:0` test case here is duplicated, additionally, you want the test to run for the CPU case

```suggestion
        torch.device("cpu"),
        pytest.param(
            torch.device("cuda", 0), marks=pytest.mark.skipif(not torch.cuda.is_available(), reason="Test requires GPU")
        ),
    ],
)
```

Reply:
Yes, I noticed it was there before, but I don't think there is any benefit in testing for both cuda:0 and cuda:1 :thinking:

Reply:
Makes Sense!